cmd/releasego/publish-announcement.go

```diff
@@ -200,23 +198,46 @@ func publishAnnouncement(p subcmd.ParseFunc) (err error) {
 		return fmt.Errorf("file %s already exists in go-devblog repository", blogFilePath)
 	}
 
-	if err := githubutil.Retry(func() error {
-		// Upload the announcement to the go-devblog repositoryy main branch with proper commit message
-		if err := githubutil.UploadFile(
-			ctx,
-			client,
-			org,
-			repo,
-			"main",
-			blogFilePath,
-			fmt.Sprintf("Add new blog post for new release in %s", releaseDate.Format("2006-01-02")),
-			content.Bytes()); err != nil {
-			return fmt.Errorf("error uploading file to go-devblog repository : %w", err)
-		}
+	// Create a feature branch (gitpr convention: dev/<purpose>/<name>) and open a PR to main.
+	prSet := gitpr.PRRefSet{Name: "main", Purpose: fmt.Sprintf("blog-%d", time.Now().Unix())}
+	branchName := prSet.PRBranch()
+
+	if err := githubutil.CreateBranch(ctx, client, org, repo, branchName, "main"); err != nil {
+		return fmt.Errorf("error creating branch %s: %w", branchName, err)
+	}
+
+	if err := githubutil.UploadFile(
+		ctx,
+		client,
+		org,
+		repo,
+		branchName,
+		blogFilePath,
+		fmt.Sprintf("Add blog post: %s", releaseInfo.Title),
+		content.Bytes()); err != nil {
+		return fmt.Errorf("error uploading file to branch %s: %w", branchName, err)
```

Comment:
Newline for readability after multi-line `if`:

```suggestion
		content.Bytes()); err != nil {

		return fmt.Errorf("error uploading file to branch %s: %w", branchName, err)
```

Reply:
Weird... filed https://github.com/microsoft/go-lab/issues/239 to undo the linter.